pipeline/docker/docker.go

```diff
@@ -0,0 +1,110 @@
+// Package docker provides a Pipe that creates and pushes a Docker image
+package docker
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/config"
+	"github.com/goreleaser/goreleaser/context"
+	"github.com/goreleaser/goreleaser/pipeline"
+	"github.com/pkg/errors"
+)
+
+// ErrNoDocker is shown when docker cannot be found in $PATH
+var ErrNoDocker = errors.New("docker not present in $PATH")
+
+// Pipe for docker
+type Pipe struct{}
+
+// Description of the pipe
+func (Pipe) Description() string {
+	return "Creating Docker images"
+}
+
+// Run the pipe
+func (Pipe) Run(ctx *context.Context) error {
+	if len(ctx.Config.Dockers) == 0 || ctx.Config.Dockers[0].Image == "" {
+		return pipeline.Skip("docker section is not configured")
+	}
+	_, err := exec.LookPath("docker")
+	if err != nil {
+		return ErrNoDocker
+	}
+	for _, docker := range ctx.Config.Dockers {
+		var imagePlatform = docker.Goos + docker.Goarch + docker.Goarm
+		for platform, groups := range ctx.Binaries {
+			if platform != imagePlatform {
+				continue
+			}
+			for folder, binaries := range groups {
+				for _, binary := range binaries {
+					if binary.Name != docker.Binary {
+						continue
+					}
+					var err = doRun(ctx, folder, docker, binary)
+					if err != nil && !pipeline.IsSkip(err) {
+						return err
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func doRun(ctx *context.Context, folder string, docker config.Docker, binary context.Binary) error {
+	var root = filepath.Join(ctx.Config.Dist, folder)
+	var dockerfile = filepath.Join(root, "Dockerfile")
```

Comment:
Would it make sense to make the name of the file (in this case `Dockerfile`) also configurable, but the default? The reason is: I saw already a few projects that owns multiple Dockerfiles. One purpose of this can be a `Dockerfile` (for prod) and a `Dockerfile.dev` (for specific dev envs with profilers, etc.)
In a `docker build -t ...` you can add the `-f Dockerfile.dev` parameter to tell Docker which Dockerfile to build

Reply:
I don't see why anyone would use a Dockerfile.dev within goreleaser... 

anyway, we can add in the future if people ask for it :D

Reply:
According to the PR description and the sample configuration, i have implied you had think of this feature:

Copy from the PR text:
```
dockers:
  - goos: linux
    goarch: amd64
    goarm: ''
    dockerfile: Dockerfile
    image: goreleaser/goreleaser
```

Or what is the use case for `dockerfile: Dockerfile`?

One project with a `Dockerfile.dev` is https://github.com/cloudflare/complainer.
And here you have a bunch of projects that have a `Dockerfile.dev`: https://github.com/search?utf8=%E2%9C%93&q=filename%3ADockerfile.dev&type=Code
To be fair, this is not only golang projects., but this shows that there is a use case for this.

Reply:
ohh, I think I get what you meant... lemme fix that real quick